SYCL/ESIMD/dp4a.cpp

```diff
@@ -0,0 +1,93 @@
+//==------------------ dp4a.cpp  - DPC++ ESIMD on-device test --------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows
+// REQUIRES: linux && gpu
+// RUN: %clangxx-esimd -fsycl %s -o %t.out
+// RUN: %HOST_RUN_PLACEHOLDER %t.out
+// RUNx: %ESIMD_RUN_PLACEHOLDER %t.out
```

Comment:
Why the test is not run (RUNx)?
if it is expected add corresponding comment  and/or XFAIL tests.

Reply:
Because this feature is not supported on SKL GPU device. I have pushed new change to add comment and mark this as expected fail. As suggested by Konst, also filed new JIRA for enabling platform specific tests.